packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -216,6 +216,14 @@ export class Pipeline extends cdk.Construct implements events.IEventRuleTarget {
     return this.stages.length;
   }
 
+  public grantBucketRead(identity?: iam.IPrincipal): void {
+    this.artifactBucket.grantRead(identity);
```

Comment:
dependency-wise, would it make sense to just expose `artifactBucket` in `IPipeline`?

Reply:
Do you think I would bother with these methods, if I could just use `artifactBucket` here? 🙂

You can't expose `artifactBucket` in `IPipeline`, as that would be a circular dependency between the `aws-codepipeline-api` and `aws-s3` modules.